infra/gcp/ensure-main-project.sh

```diff
@@ -167,6 +173,48 @@ function ensure_terraform_state_buckets() {
     done
 }
 
+function ensure_prow_buckets() {
+    if [ $# -ne 1 ] || [ -z "$1" ]; then
+        echo "${FUNCNAME[0]}(gcp_project) requires 1 argument" >&2
+        return 1
+    fi
+
+    local project="${1}"
+
+    for bucket in "${PROW_BUCKETS[@]}"; do
+        local svc_acct_name="${bucket}-sa"
+        local svc_acct_email="$(svc_acct_email "${project}" \
+        "${svc_acct_name}")"
+        local SECRET_ID="${svc_acct_name}-key"
+
+        color 6 "Ensuring bucket ${bucket} exists and is only word-readable"
+        ensure_public_gcs_bucket "${project}" "gs://${bucket}"
+
+
+        color 6 "Creating service account: ${svc_acct_name}"
+        ensure_service_account \
+            "${project}" \
+            "${svc_acct_name}" \
+            "${svc_acct_name}"
+
+        color 6 "Empowering service account: ${svc_acct_name}"
+        empower_svcacct_to_write_gcs_bucket "${svc_acct_email}" "gs://${bucket}"
+
+        color 6 "Ensure secret ${SECRET_ID} exists in project ${PROJECT}"
+        ensure_secret "${project}" "${SECRET_ID}"
+
+        color "Ensure ${SECRET_ID} contains secret key for ${svc_acct_name}"
+        ensure_serviceaccount_key_secret "${project}" "${SECRET_ID}" "${svc_acct_email}"
+
+        color 6 "Empowering k8s-infra-prow-oncall@kubernetes.io to read secret ${SECRET_ID}"
+        ensure_secrets_role_binding \
+            "projects/${project}/secrets/${SECRET_ID}" \
+            "group:k8s-infra-prow-oncall@kubernetes.io" \
+            "roles/secretmanager.secretAccessor"
```

Comment:
This is basically https://github.com/kubernetes/k8s.io/blob/4d98e1c294c6e6158eca1efb9e6a4f8c3003a572/infra/gcp/ensure-conformance-storage.sh#L104-L131

You could parameterize it on project and pull into lib_iam.sh for re-use.  Can be done as followup though